src/client.ts

```diff
@@ -1116,11 +1116,19 @@ export class StreamChat<
       logger: this.logger,
     });
 
+    let warmUpPromise;
     if (this.options.warmUp) {
-      this.doAxiosRequest('options', this.baseURL + '/connect');
+      warmUpPromise = this.doAxiosRequest('options', this.baseURL + '/connect');
```

Comment:
The start1, start2, wait2, wait1 logic looks a bit tricky to follow. Maybe something like the following would make it clearer:

```js
const warmUpHttp = () => this.options.warmUp ? this.doAxiosRequest('options', this.baseURL + '/connect') : Promise.resolve();
const [_, handshake] = await Promise.all([warmUpHttp(), this.wsConnection.connect()]);
```

Reply:
Already merged. I was too slow x)

Reply:
good points but `Promise.all` would make bundle bigger.